src/providers/workers-ai/chatComplete.ts

```diff
@@ -0,0 +1,189 @@
+    {
+      param: 'messages',
+      required: true,
+      transform: (params: Params) => {
+        let messages: Message[] = [];
+        // Transform the chat messages into a simple prompt
+        if (!!params.messages) {
+          params.messages.forEach((msg) => {
+            if (
+              msg.content &&
+              typeof msg.content === 'object' &&
+              msg.content.length
+            ) {
+              const transformedMessage: Record<string, any> = {
+                role: msg.role,
+                content: [],
+              };
+              msg.content.forEach((item) => {
+                if (item.type === 'text') {
+                  transformedMessage.content.push({
+                    type: item.type,
+                    text: item.text,
+                  });
+                } else if (
+                  item.type === 'image_url' &&
+                  item.image_url &&
+                  item.image_url.url
+                ) {
+                  const parts = item.image_url.url.split(';');
+                  if (parts.length === 2) {
+                    const base64ImageParts = parts[1].split(',');
+                    const base64Image = base64ImageParts[1];
+                    const mediaTypeParts = parts[0].split(':');
+                    if (mediaTypeParts.length === 2 && base64Image) {
+                      const mediaType = mediaTypeParts[1];
+                      transformedMessage.content.push({
+                        type: 'image',
+                        source: {
+                          type: 'base64',
+                          media_type: mediaType,
+                          data: base64Image,
+                        },
+                      });
+                    }
+                  }
+                }
+              });
+              messages.push(transformedMessage as Message);
+            } else {
+              messages.push({
+                role: msg.role,
+                content: msg.content,
+              });
+            }
+          });
+        }
+
+        return messages;
+      },
+    },
+  ],
+  stream: {
+    param: 'stream',
+    default: false,
+  },
+};
+
+export interface WorkersAiErrorObject {
+  type: string;
+  message: string;
+}
+
+interface WorkersAiErrorResponse {
+  type: string;
+  error: WorkersAiErrorObject;
+}
+
+interface WorkersAiChatCompleteResponse {
+  result: {
+    response: string;
+  };
+  success: boolean;
+  errors: string[];
+  messages: string[];
+}
+
+interface WorkersAiChatCompleteStreamResponse {
+  response: string;
+  p?: string;
+}
+
+// TODO: cloudflare do not return the usage
+// TODO: return the model
+export const WorkersAiChatCompleteResponseTransform: (
+  response: WorkersAiChatCompleteResponse | WorkersAiErrorResponse,
+  responseStatus: number
+) => ChatCompletionResponse | ErrorResponse = (response, responseStatus) => {
+  if (responseStatus !== 200 && 'errors' in response) {
+    return {
```

Comment:
In one of the recent release, we pushed some refactoring for error handling on provider level. Instead of writing same code for all provider methods, you can now just define a error handler and use it in each transformer. You can check anthropic chatComplete.ts for an example.